File: listings/while_example.m
% while ( CONDITION ) 
%   STATEMENT
% end.
% 
% EXAMPLE: Tell me when a new minute starts
%
clc;            %clear screen
c=clock;        %get time vector

% 6th element of c is seconds
while ( c(6) < 59.9)
    c=clock;
end
disp('start new minute of your life');

File: listings/while_example.py
# while CONDITION:
#   STATEMENT
#   [if CONDITION:
#       break]
# 
# EXAMPLE: Run until a new minute starts

import time, datetime

#infinite loop, unless there's a break
#somewhere
while True:
    #use datetime module to get second
    sec = datetime.datetime.now().second
    
    print(sec)
    time.sleep(1)

    #not every while needs a break
    if sec == 59:
        break
    
	
